pystapi-client/src/pystapi_client/client.py

```diff
@@ -0,0 +1,283 @@
+            request_modifier=request_modifier,
+            timeout=timeout,
+        )
+
+        client.read_links()
+        client.read_conformance()
+
+        if not client.has_conforms_to():
+            warnings.warn(NoConformsTo())
+
+        return client
+
+    def get_single_link(
+        self,
+        rel: str | None = None,
+        media_type: str | Iterable[str] | None = None,
+    ) -> Link | None:
+        """Get a single :class:`~stapi_pydantic.Link` instance associated with this object.
+
+        Args:
+            rel : If set, filter links such that only those
+                matching this relationship are returned.
+            media_type: If set, filter the links such that only
+                those matching media_type are returned. media_type can
+                be a single value or a list of values.
+
+        Returns:
+            :class:`~stapi_pydantic.Link` | None: First link that matches ``rel``
+                and/or ``media_type``, or else the first link associated with
+                this object.
+        """
+        if rel is None and media_type is None:
+            return next(iter(self._extra_fields["links"]), None)
+        if media_type and isinstance(media_type, str):
+            media_type = [media_type]
+        return next(
+            (
+                link
+                for link in self._extra_fields["links"]
+                if (rel is None or link.rel == rel) and (media_type is None or link.media_type in media_type)
+            ),
+            None,
+        )
+
+    def read_links(self) -> None:
+        """ """
+        links = self._stapi_io._read_json("/").get("links", [])
+        if links:
+            self._extra_fields["links"] = [Link(**link) for link in links]
+        else:
+            warnings.warn("No links found in the root of the STAPI API")
+            self._extra_fields["links"] = [
+                Link(
+                    href=urllib.parse.urljoin(str(self._stapi_io.root_url), link["endpoint"]),
+                    rel=link["rel"],
+                    method=link["method"],
+                )
+                for link in DEFAULT_LINKS
+            ]
+
+    def read_conformance(self) -> None:
+        conformance: list[str] = []
+        for endpoint in ["/conformance", "/"]:
+            try:
+                conformance = self._stapi_io._read_json("conformance").get("conformsTo", [])
+                break
+            except APIError:
+                continue
+
+        if conformance:
+            self.set_conforms_to(conformance)
+
+    def has_conforms_to(self) -> bool:
+        """Whether server contains list of ``"conformsTo"`` URIs"""
+        return "conformsTo" in self._extra_fields
+
+    def get_conforms_to(self) -> list[str]:
+        """List of ``"conformsTo"`` URIs
+
+        Return:
+            List[str]: List of  URIs that the server conforms to
+        """
+        return cast(list[str], self._extra_fields.get("conformsTo", []).copy())
```

Comment:
Any reason not to make `conforms_to` a member attribute of `Client`?

Reply:
No reason. We were just following pystac patterns. We moved it now.